framework/src/auxkernels/BuildArrayVariableAux.C

```diff
@@ -0,0 +1,81 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "BuildArrayVariableAux.h"
+
+#include "SystemBase.h"
+
+registerMooseObject("MooseApp", BuildArrayVariableAux);
+
+defineLegacyParams(BuildArrayVariableAux);
+
+InputParameters
+BuildArrayVariableAux::validParams()
+{
+  InputParameters params = ArrayAuxKernel::validParams();
+  params.addParam<std::vector<VariableName>>(
+      "component_variables",
+      "The variables that make up each component of the output array variable.");
+  params.addClassDescription("Copy multiple variables into the components of an array variable.");
+  return params;
+}
+
+BuildArrayVariableAux::BuildArrayVariableAux(const InputParameters & parameters)
+  : ArrayAuxKernel(parameters)
+{
+  const auto & var_names = parameters.get<std::vector<VariableName>>("component_variables");
+
+  // Check the number of component variables
+  if (var_names.size() != _var.count())
+    mooseError("The array variable has ",
+               _var.count(),
+               " components, but ",
+               var_names.size(),
+               " component variables were specified.");
+
+  // Get pointers to each of the component vars
+  THREAD_ID tid = parameters.get<THREAD_ID>("_tid");
+  for (VariableName name : var_names)
+    _component_vars.push_back(&_subproblem.getVariable(tid, name));
+
+  // Check the FEType of the output variable
+  if ((_var.feType().family != FEFamily::MONOMIAL) || (_var.feType().order != Order::CONSTANT))
+    mooseError("BuildArrayVariableAux only supports constant monomial variables");
+
+  // Check the FEType of the component vars
+  for (auto var : _component_vars)
+  {
+    if ((var->feType().family != FEFamily::MONOMIAL) || (var->feType().order != Order::CONSTANT))
+      mooseError("BuildArrayVariableAux only supports constant monomial variables");
+  }
+}
+
+RealEigenVector
+BuildArrayVariableAux::computeValue()
```

Comment:
Why not directly override `compute` function? If you do so, we can just ask for the same shape functions for all the component variables and the array variable and the shape function does not have to be constant MONOMIAL.

Reply:
Then you would have to reimplement all the nodal vs elemental handling stuff in compute.  Seems like the wrong approach to me.

Reply:
We possibly do not need to check `isNodal()` for this particular aux kernel. `cholesky_solve` is not cheap and not necessary here.

Reply:
I'm not sure what you're suggesting.  Checking isNodal is trivial computationally.  Also, for const monomial elemental vars - we don't do the cholesky solve already - only for higher order ones.  So this kernel would run just fine without overriding compute.

Reply:
What I mean is to directly couple all DoF values and directly assign the array variable DoF values with them. This can be done for any type of variables and the variable fe type does not have to be constant monomial. I do have a strong opinion on this.